### DISCUSSION TITLE
IClientStartup and IServerStartup not executed

### DISCUSSION THREAD

**Discussion by LuisVilaplana** [oqtane/oqtane.framework#3891] [General]
Hello, why aren't the IClientStartup and IServerStartup classes in my external modules not executing? On the client side, for example, I created a folder named Startup and a file named ClientStartup.cs, defined as follows:

using Microsoft.Extensions.DependencyInjection;
using Oqtane.Services;
//using Syncfusion.Blazor;

namespace ProsoftPOS.Module.FacturaElectronica.Client.Startup
{

    public class ClientStartup : IClientStartup
    {
        public void ConfigureServices(IServiceCollection services)
        {

        }
    }
}

But it never gets executed, neither does ServerStartup. I've tried with a fresh installation of Oqtane and with an new external module created using the Module Creator. I'm using version 5.0.2 of Oqtane.

Regards

**Comment by sbwalker** [comment]
@LuisVilaplana nothing has changed in recent releases which would affect the startup interfaces. Is it possible that you are using dependencies in your startup implementations which are not being deployed to the Oqtane /bin folder as part of the build process of your module? If Oqtane encounters missing dependencies in the run-time environment, it will not be able to execute your startup logic.

**Comment by leigh-pointer** [comment]
@LuisVilaplana As @sbwalker said but to add. If you are using a dependency or dependencies its a good idea to add

	<CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>

to the **PropertyGroup** inside your client project, and also update the debug.cmd file in the Packages project to XCOPY the dependency or dependencies to the oqtane.server bin

XCOPY "..\Client\bin\Debug\net8.0\[**TheDependency.dll**]" "..\..\oqtane.framework\Oqtane.Server\bin\Debug\net8.0\" /Y

**Comment by sbwalker** [comment]
@LuisVilaplana did the answers above provide a solution to your problem? Or are you still unable to get the startup interfaces to execute?

**Reply by LuisVilaplana** [reply]
Apologies for the delay, I haven't noticed any missing dependencies. I will review and keep you posted. I'm also going to apply  @leigh-pointer's suggestion.
Thanks

**Reply by LuisVilaplana** [reply]
I have checked for any dependencies missing from the debug.cmd of my module, but I couldn't find any. Is there any way to see if there's an error of this type? I can't seem to locate the source of the problem.
